cognite/client/_api/geospatial.py

```diff
@@ -0,0 +1,528 @@
+    FeatureAggregateList,
+    FeatureList,
+    FeatureType,
+    FeatureTypeList,
+    FeatureTypeUpdate,
+    OrderSpec,
+)
+
+
+class GeospatialAPI(APIClient):
+
+    _RESOURCE_PATH = "/geospatial"
+
+    @staticmethod
+    def _feature_resource_path(feature_type: FeatureType):
+        return f"{GeospatialAPI._RESOURCE_PATH}/featuretypes/{feature_type.external_id}/features"
+
+    def create_feature_types(
+        self, feature_type: Union[FeatureType, List[FeatureType]]
+    ) -> Union[FeatureType, FeatureTypeList]:
+        """`Creates feature types`
+        <https://pr-1323.specs.preview.cogniteapp.com/v1.json.html#operation/createFeatureTypes>
+
+        Args:
+            feature_type (Union[FeatureType, List[FeatureType]]): feature type definition or list of feature type definitions to create.
+
+        Returns:
+            Union[FeatureType, FeatureTypeList]: Created feature type definition(s)
+
+        Examples:
+
+            Create new type definitions:
+
+                >>> from cognite.experimental import CogniteClient
+                >>> from cognite.experimental.data_classes.geospatial import FeatureType
+                >>> c = CogniteClient()
+                >>> feature_types = [
+                ...     FeatureType(external_id="wells", properties={"location": {"type": "POINT", "srid": 4326}})
+                ...     FeatureType(external_id="pipelines", properties={"location": {"type": "LINESTRING", "srid": 2001}})
+                ... ]
+                >>> res = c.geospatial.create_feature_types(feature_types)
+        """
+        return self._create_multiple(
+            items=feature_type, cls=FeatureTypeList, resource_path=f"{self._RESOURCE_PATH}/featuretypes"
+        )
+
+    def delete_feature_types(self, external_id: Union[str, List[str]], recursive: bool = False) -> None:
+        """`Delete one or more feature type`
+        <https://pr-1323.specs.preview.cogniteapp.com/v1.json.html#operation/deleteFeatureTypes>
+
+        Args:
+            external_id (Union[str, List[str]]): External ID or list of external ids
+            recursive (bool): if `true` the features will also be dropped
+
+        Returns:
+            None
+
+        Examples:
+
+            Delete feature type definitions external id:
+
+                >>> from cognite.experimental import CogniteClient
+                >>> c = CogniteClient()
+                >>> c.geospatial.delete_feature_types(external_id=["wells", "pipelines"])
+        """
+        extra_body_fields = {"recursive": True} if recursive else {}
+        return self._delete_multiple(
+            external_ids=external_id,
+            wrap_ids=True,
+            resource_path=f"{self._RESOURCE_PATH}/featuretypes",
+            extra_body_fields=extra_body_fields,
+        )
+
+    def list_feature_types(self) -> FeatureTypeList:
+        """`List feature types`
+        <https://pr-1323.specs.preview.cogniteapp.com/v1.json.html#operation/listFeatureTypes>
```

Comment:
should we update this link as well?

Reply:
Good point! The link will be updated when the api spec merged has happened.

Reply:
links updated